### PR TITLE
More promote_op fixes

### PR DESCRIPTION
**Pull request by p-zubieta**
@nalimilan 

Note: This PR is against branch `nl/promote_op` instead of `master`

### CODE REVIEW

File: test/arrayops.jl [hunk 1]
```diff
@@ -1418,17 +1418,14 @@ module RetTypeDecl
...
-
-    @test @inferred(m+[m,m]) == [m+m,m+m]
-    @test @inferred([m,m]+m) == [m+m,m+m]
-    @test @inferred(m.*[m,m]) == [m2,m2]
-    @test @inferred([m,m].*m) == [m2,m2]
-    @test @inferred([m 2m; m m]*[m,m]) == [3m2,2m2]
-    @test @inferred([m m].*[m,m]) == [m2 m2; m2 m2]
+    convert{T<:Real,pow}(::Type{MeterUnits{T,pow}}, y::Real) = MeterUnits{T,pow}(convert(T,y))
+
+    @test m+[m,m] == [m+m,m+m]
```
Comment (nalimilan) [inline test/arrayops.jl]: Why remove the `inferred` calls?

**Comment by nalimilan** [comment]
Thanks! Though @JeffBezanson is the one to ask...

Ref: https://github.com/JuliaLang/julia/pull/17114

File: test/arrayops.jl (hunk 1 above)
Comment (pabloferz) [inline test/arrayops.jl]: I made a mistake while copying some of the code I tested locally. Is fixed now.

**Comment by nalimilan** [comment]
I've picked your commit into https://github.com/JuliaLang/julia/pull/16995, let's continue there.

**Pull request closed without merging**